examples/dev-example/src/node/model-server-client.ts

```diff
@@ -8,46 +8,44 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR MIT
  *******************************************************************************/
-import { AnyObject, MessageDataMapper, MessageType, SubscriptionOptions } from '@eclipse-emfcloud/modelserver-client';
-import { TheiaBackendModelServerClient } from '@eclipse-emfcloud/modelserver-theia/lib/node';
+import {
+    AnyObject,
+    MessageDataMapper,
+    MessageType,
+    SubscriptionListener,
+    SubscriptionOptionsV2
+} from '@eclipse-emfcloud/modelserver-client';
+import { TheiaBackendModelServerClientV2 } from '@eclipse-emfcloud/modelserver-theia/lib/node';
 import { injectable } from '@theia/core/shared/inversify';
+import URI from 'urijs';
 
 import { DevModelServerClient } from '../common/dev-model-server-client';
 
 @injectable()
-export class CustomDevModelServerClient extends TheiaBackendModelServerClient implements DevModelServerClient {
+export class CustomDevModelServerClient extends TheiaBackendModelServerClientV2 implements DevModelServerClient {
     private intervalId: NodeJS.Timeout;
 
-    subscribe(modeluri: string, options: SubscriptionOptions = {}): void {
+    subscribe(modeluri: URI, listener?: SubscriptionListener, options: SubscriptionOptionsV2 = {}): SubscriptionListener {
         if (options.timeout) {
             this.setKeepAliveInterval(modeluri, options.timeout);
         }
-        return super.subscribe(modeluri, options);
+        return super.subscribe(modeluri, listener, options);
     }
 
-    private setKeepAliveInterval(modelUri: string, timeout: number): void {
-        if (!this.isSocketOpen(modelUri) && modelUri === 'Coffee.ecore') {
+    private setKeepAliveInterval(modeluri: URI, timeout: number): void {
+        if (!this.isSocketOpen(modeluri) && modeluri.toString() === 'Coffee.ecore') {
             this.intervalId = setInterval(
-                () => this.send(modelUri, { type: MessageType.keepAlive, data: undefined }),
-                timeout > 1000 ? timeout - 1000 : 1
+                () => this.send(modeluri, { type: MessageType.keepAlive, data: '' }),
```

Comment:
why do we send an empty string as data instead of `undefined` now?

Reply:
Good catch thanks, this is an leftover can be reverted again.